commit b5621b66e778112c78d381414dff8917320f71d2
Author: Peter Eisentraut <peter@eisentraut.org>
Date:   Tue Nov 8 18:45:29 2022 +0100

    Unify some internal error message wordings

diff --git a/src/backend/catalog/aclchk.c b/src/backend/catalog/aclchk.c
index aa5a2ed9483..42360d37ca5 100644
--- a/src/backend/catalog/aclchk.c
+++ b/src/backend/catalog/aclchk.c
@@ -1243,11 +1243,11 @@ SetDefaultACL(InternalDefaultACL *iacls)
 			if (iacls->all_privs && this_privileges == ACL_NO_RIGHTS)
 				this_privileges = ACL_ALL_RIGHTS_SCHEMA;
 			break;
 
 		default:
-			elog(ERROR, "unrecognized objtype: %d",
+			elog(ERROR, "unrecognized object type: %d",
 				 (int) iacls->objtype);
 			objtype = 0;		/* keep compiler quiet */
 			break;
 	}
 
@@ -3588,11 +3588,11 @@ aclcheck_error(AclResult aclerr, ObjectType objtype,
 					case OBJECT_TRANSFORM:
 					case OBJECT_TRIGGER:
 					case OBJECT_TSPARSER:
 					case OBJECT_TSTEMPLATE:
 					case OBJECT_USER_MAPPING:
-						elog(ERROR, "unsupported object type %d", objtype);
+						elog(ERROR, "unsupported object type: %d", objtype);
 				}
 
 				ereport(ERROR,
 						(errcode(ERRCODE_INSUFFICIENT_PRIVILEGE),
 						 errmsg(msg, objectname)));
@@ -3726,11 +3726,11 @@ aclcheck_error(AclResult aclerr, ObjectType objtype,
 					case OBJECT_ROLE:
 					case OBJECT_TRANSFORM:
 					case OBJECT_TSPARSER:
 					case OBJECT_TSTEMPLATE:
 					case OBJECT_USER_MAPPING:
-						elog(ERROR, "unsupported object type %d", objtype);
+						elog(ERROR, "unsupported object type: %d", objtype);
 				}
 
 				ereport(ERROR,
 						(errcode(ERRCODE_INSUFFICIENT_PRIVILEGE),
 						 errmsg(msg, objectname)));
@@ -3826,11 +3826,11 @@ pg_aclmask(ObjectType objtype, Oid table_oid, AttrNumber attnum, Oid roleid,
 			/* not reached, but keep compiler quiet */
 			return ACL_NO_RIGHTS;
 		case OBJECT_TYPE:
 			return pg_type_aclmask(table_oid, roleid, mask, how);
 		default:
-			elog(ERROR, "unrecognized objtype: %d",
+			elog(ERROR, "unrecognized object type: %d",
 				 (int) objtype);
 			/* not reached, but keep compiler quiet */
 			return ACL_NO_RIGHTS;
 	}
 }
diff --git a/src/backend/catalog/objectaddress.c b/src/backend/catalog/objectaddress.c
index 284ca55469e..c7de7232b89 100644
--- a/src/backend/catalog/objectaddress.c
+++ b/src/backend/catalog/objectaddress.c
@@ -1162,11 +1162,11 @@ get_object_address(ObjectType objtype, Node *object,
 				address.objectId = get_statistics_object_oid(castNode(List, object),
 															 missing_ok);
 				address.objectSubId = 0;
 				break;
 			default:
-				elog(ERROR, "unrecognized objtype: %d", (int) objtype);
+				elog(ERROR, "unrecognized object type: %d", (int) objtype);
 				/* placate compiler, in case it thinks elog might return */
 				address.classId = InvalidOid;
 				address.objectId = InvalidOid;
 				address.objectSubId = 0;
 		}
@@ -1353,11 +1353,11 @@ get_object_address_unqualified(ObjectType objtype,
 			address.classId = SubscriptionRelationId;
 			address.objectId = get_subscription_oid(name, missing_ok);
 			address.objectSubId = 0;
 			break;
 		default:
-			elog(ERROR, "unrecognized objtype: %d", (int) objtype);
+			elog(ERROR, "unrecognized object type: %d", (int) objtype);
 			/* placate compiler, which doesn't know elog won't return */
 			address.classId = InvalidOid;
 			address.objectId = InvalidOid;
 			address.objectSubId = 0;
 	}
@@ -1430,11 +1430,11 @@ get_relation_by_qualified_name(ObjectType objtype, List *object,
 						(errcode(ERRCODE_WRONG_OBJECT_TYPE),
 						 errmsg("\"%s\" is not a foreign table",
 								RelationGetRelationName(relation))));
 			break;
 		default:
-			elog(ERROR, "unrecognized objtype: %d", (int) objtype);
+			elog(ERROR, "unrecognized object type: %d", (int) objtype);
 			break;
 	}
 
 	/* Done. */
 	address.objectId = RelationGetRelid(relation);
@@ -1506,11 +1506,11 @@ get_object_address_relobject(ObjectType objtype, List *object,
 				get_relation_policy_oid(reloid, depname, missing_ok) :
 				InvalidOid;
 			address.objectSubId = 0;
 			break;
 		default:
-			elog(ERROR, "unrecognized objtype: %d", (int) objtype);
+			elog(ERROR, "unrecognized object type: %d", (int) objtype);
 	}
 
 	/* Avoid relcache leak when object not found. */
 	if (!OidIsValid(address.objectId))
 	{
@@ -1698,11 +1698,11 @@ get_object_address_opcf(ObjectType objtype, List *object, bool missing_ok)
 			address.classId = OperatorFamilyRelationId;
 			address.objectId = get_opfamily_oid(amoid, object, missing_ok);
 			address.objectSubId = 0;
 			break;
 		default:
-			elog(ERROR, "unrecognized objtype: %d", (int) objtype);
+			elog(ERROR, "unrecognized object type: %d", (int) objtype);
 			/* placate compiler, which doesn't know elog won't return */
 			address.classId = InvalidOid;
 			address.objectId = InvalidOid;
 			address.objectSubId = 0;
 	}
@@ -1816,11 +1816,11 @@ get_object_address_opf_member(ObjectType objtype,
 					ReleaseSysCache(tp);
 				}
 			}
 			break;
 		default:
-			elog(ERROR, "unrecognized objtype: %d", (int) objtype);
+			elog(ERROR, "unrecognized object type: %d", (int) objtype);
 	}
 
 	return address;
 }
 
diff --git a/src/backend/commands/alter.c b/src/backend/commands/alter.c
index 55219bb0974..5dabba4dd06 100644
--- a/src/backend/commands/alter.c
+++ b/src/backend/commands/alter.c
@@ -97,11 +97,11 @@ report_name_conflict(Oid classId, const char *name)
 			break;
 		case SubscriptionRelationId:
 			msgfmt = gettext_noop("subscription \"%s\" already exists");
 			break;
 		default:
-			elog(ERROR, "unsupported object class %u", classId);
+			elog(ERROR, "unsupported object class: %u", classId);
 			break;
 	}
 
 	ereport(ERROR,
 			(errcode(ERRCODE_DUPLICATE_OBJECT),
@@ -140,11 +140,11 @@ report_namespace_conflict(Oid classId, const char *name, Oid nspOid)
 		case TSConfigRelationId:
 			Assert(OidIsValid(nspOid));
 			msgfmt = gettext_noop("text search configuration \"%s\" already exists in schema \"%s\"");
 			break;
 		default:
-			elog(ERROR, "unsupported object class %u", classId);
+			elog(ERROR, "unsupported object class: %u", classId);
 			break;
 	}
 
 	ereport(ERROR,
 			(errcode(ERRCODE_DUPLICATE_OBJECT),
diff --git a/src/backend/utils/adt/acl.c b/src/backend/utils/adt/acl.c
index 4fac402e5b6..0bc79cba2b1 100644
--- a/src/backend/utils/adt/acl.c
+++ b/src/backend/utils/adt/acl.c
@@ -805,11 +805,11 @@ acldefault(ObjectType objtype, Oid ownerId)
 		case OBJECT_PARAMETER_ACL:
 			world_default = ACL_NO_RIGHTS;
 			owner_default = ACL_ALL_RIGHTS_PARAMETER_ACL;
 			break;
 		default:
-			elog(ERROR, "unrecognized objtype: %d", (int) objtype);
+			elog(ERROR, "unrecognized object type: %d", (int) objtype);
 			world_default = ACL_NO_RIGHTS;	/* keep compiler quiet */
 			owner_default = ACL_NO_RIGHTS;
 			break;
 	}
 
@@ -902,11 +902,11 @@ acldefault_sql(PG_FUNCTION_ARGS)
 			break;
 		case 'T':
 			objtype = OBJECT_TYPE;
 			break;
 		default:
-			elog(ERROR, "unrecognized objtype abbreviation: %c", objtypec);
+			elog(ERROR, "unrecognized object type abbreviation: %c", objtypec);
 	}
 
 	PG_RETURN_ACL_P(acldefault(objtype, owner));
 }
 
