commit 45d5ecab49f6c0df607d3ca2ce1995229f416d1a
Author: Peter Eisentraut <peter@eisentraut.org>
Date:   Sun Nov 13 21:09:09 2022 +0100

    libpq: Add missing newlines to error messages

diff --git a/src/interfaces/libpq/fe-protocol3.c b/src/interfaces/libpq/fe-protocol3.c
index f001137b769..f5664e0f831 100644
--- a/src/interfaces/libpq/fe-protocol3.c
+++ b/src/interfaces/libpq/fe-protocol3.c
@@ -200,11 +200,11 @@ pqParseInput3(PGconn *conn)
 						conn->result = PQmakeEmptyPGresult(conn,
 														   PGRES_COMMAND_OK);
 						if (!conn->result)
 						{
 							appendPQExpBufferStr(&conn->errorMessage,
-												 libpq_gettext("out of memory"));
+												 libpq_gettext("out of memory\n"));
 							pqSaveErrorResult(conn);
 						}
 					}
 					if (conn->result)
 						strlcpy(conn->result->cmdStatus, conn->workBuffer.data,
@@ -225,11 +225,11 @@ pqParseInput3(PGconn *conn)
 						conn->result = PQmakeEmptyPGresult(conn,
 														   PGRES_PIPELINE_SYNC);
 						if (!conn->result)
 						{
 							appendPQExpBufferStr(&conn->errorMessage,
-												 libpq_gettext("out of memory"));
+												 libpq_gettext("out of memory\n"));
 							pqSaveErrorResult(conn);
 						}
 						else
 						{
 							conn->pipelineStatus = PQ_PIPELINE_ON;
@@ -254,11 +254,11 @@ pqParseInput3(PGconn *conn)
 						conn->result = PQmakeEmptyPGresult(conn,
 														   PGRES_EMPTY_QUERY);
 						if (!conn->result)
 						{
 							appendPQExpBufferStr(&conn->errorMessage,
-												 libpq_gettext("out of memory"));
+												 libpq_gettext("out of memory\n"));
 							pqSaveErrorResult(conn);
 						}
 					}
 					conn->asyncStatus = PGASYNC_READY;
 					break;
@@ -272,11 +272,11 @@ pqParseInput3(PGconn *conn)
 							conn->result = PQmakeEmptyPGresult(conn,
 															   PGRES_COMMAND_OK);
 							if (!conn->result)
 							{
 								appendPQExpBufferStr(&conn->errorMessage,
-													 libpq_gettext("out of memory"));
+													 libpq_gettext("out of memory\n"));
 								pqSaveErrorResult(conn);
 							}
 						}
 						conn->asyncStatus = PGASYNC_READY;
 					}
@@ -353,11 +353,11 @@ pqParseInput3(PGconn *conn)
 							conn->result = PQmakeEmptyPGresult(conn,
 															   PGRES_COMMAND_OK);
 							if (!conn->result)
 							{
 								appendPQExpBufferStr(&conn->errorMessage,
-													 libpq_gettext("out of memory"));
+													 libpq_gettext("out of memory\n"));
 								pqSaveErrorResult(conn);
 							}
 						}
 						conn->asyncStatus = PGASYNC_READY;
 					}
