commit d71e6055e45318b5cff2094025d641b954517fa9
Author: Peter Eisentraut <peter@eisentraut.org>
Date:   Thu Aug 24 14:22:02 2023 +0200

    Fix lack of message pluralization

diff --git a/src/backend/replication/slot.c b/src/backend/replication/slot.c
index 1dc27264f61..bb09c4010f8 100644
--- a/src/backend/replication/slot.c
+++ b/src/backend/replication/slot.c
@@ -1261,15 +1261,22 @@ ReportSlotInvalidation(ReplicationSlotInvalidationCause cause,
 	initStringInfo(&err_detail);
 
 	switch (cause)
 	{
 		case RS_INVAL_WAL_REMOVED:
-			hint = true;
-			appendStringInfo(&err_detail, _("The slot's restart_lsn %X/%X exceeds the limit by %llu bytes."),
-							 LSN_FORMAT_ARGS(restart_lsn),
-							 (unsigned long long) (oldestLSN - restart_lsn));
-			break;
+			{
+				unsigned long long ex = oldestLSN - restart_lsn;
+
+				hint = true;
+				appendStringInfo(&err_detail,
+								 ngettext("The slot's restart_lsn %X/%X exceeds the limit by %llu byte.",
+										  "The slot's restart_lsn %X/%X exceeds the limit by %llu bytes.",
+										  ex),
+								 LSN_FORMAT_ARGS(restart_lsn),
+								 ex);
+				break;
+			}
 		case RS_INVAL_HORIZON:
 			appendStringInfo(&err_detail, _("The slot conflicted with xid horizon %u."),
 							 snapshotConflictHorizon);
 			break;
 
